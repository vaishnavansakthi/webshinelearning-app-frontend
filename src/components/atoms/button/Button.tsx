import { IButtonProps } from "../../../types/button"

const Button = ({ text, color, size, disabled, varient, textColor = "text-white", icon, className, bgColor }: IButtonProps) => {
  return (
    <>
      <button
        className={`
        ${color === "primary"
            ? "bg-blue-500 hover:bg-blue-600"
            : color === "secondary"
              ? "bg-orange-400 hover:bg-orange-500"
              : color === "success"
                ? "bg-green-400 hover:bg-green-500"
                : color === "error"
                  ? "bg-red-500 hover:bg-red-600"
                  : `${bgColor}`
          }
         ${varient === "contained" ? "p-4 border-[1px] rounded-md" : varient === "outline" ? "p-4 border-[1px] border-gray-200 shadow-inner rounded-md" : ""}
         ${size === "large" ? "text-lg p-4" : size === "medium" ? "text-md p-2" : "text-sm p-1"}
         ${textColor} w-full font-medium text-lg rounded-lg px-6 py-2.5 text-center md:mr-5 mb-3 md:mb-0 inline-flex items-center justify-center tracking-wide ${className}
        `}
        disabled={disabled}
      >
        {icon && icon}
        {text}
      </button>
    </>
  )
}

export default Button
