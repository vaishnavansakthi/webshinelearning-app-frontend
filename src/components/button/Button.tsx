/**
 * Button Component
 * @return {React.Component}
 */
const Button = () => {
  return <div>Button</div>
}

export default Button
