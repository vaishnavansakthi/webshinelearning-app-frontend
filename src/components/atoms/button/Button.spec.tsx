import renderer from 'react-test-renderer';
import Button from './Button'; 

describe('Button component', () => {
  it('renders a button with provided text', () => {
    const tree = renderer.create(<Button type='submit' text="Click me" className='text-black' color='primary' disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
  