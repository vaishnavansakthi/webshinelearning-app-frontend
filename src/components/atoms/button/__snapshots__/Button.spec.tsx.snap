// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders a button with provided text 1`] = `
<button
  className="
        bg-blue-500 hover:bg-blue-600
         
         text-sm p-1
         text-white w-full font-medium text-lg rounded-lg px-6 py-2.5 text-center md:mr-5 mb-3 md:mb-0 inline-flex items-center justify-center tracking-wide text-black
        "
  disabled={true}
>
  Click me
</button>
`;
